### DISCUSSION TITLE
Encoding a full URL

### DISCUSSION THREAD

**Discussion by gramian** [curl/curl#9900] [Q&A] [answered]
Dear Community,

I would like to ask if `curl` can encode a full URL? Here is an example of what I would want to pass:
```
http://this.is.my/url which is not #$%^<"> encoded
```
Does `curl` have an option for this setting? I only found the `--data-urlencode` option, which to my understanding does not apply here.

Thank you for your help

**Comment by bagder** [comment, marked as answer]
It cannot "encode a full URL" because a URL is only a URL when it already is URL encoded. You want to magically convert a string that is not a URL into a URL. The libcurl URL parser can actually do this at least to some extent (because I guess you mean that it should convert non-URL parts into their corresponding counter-parts, like converting space to %20 etc), but there's no access to that feature directly from the curl tool and the given URL argument.

**Reply by gramian** [reply]
Thank you for your answer.
